nautobot/ipam/models.py

```diff
@@ -1065,13 +1065,17 @@ def _deconstruct_address(self, address):
     natural_key_field_names = ["parent__namespace", "host"]
 
     def _get_closest_parent(self):
+        if closes_parent := getattr(self, "_closest_parent", None):
+            return closes_parent
         try:
-            return (
+            closes_parent = (
                 Prefix.objects.filter(namespace=self._namespace)
                 # 3.0 TODO: disallow IPAddress from parenting to a TYPE_POOL prefix, instead pick closest TYPE_NETWORK
                 # .exclude(type=choices.PrefixTypeChoices.TYPE_POOL)
                 .get_closest_parent(self.host, include_self=True)
             )
+            setattr(self, "_closest_parent", closes_parent)
+            return closes_parent
```

Comment:
```suggestion
            closest_parent = (
                Prefix.objects.filter(namespace=self._namespace)
                # 3.0 TODO: disallow IPAddress from parenting to a TYPE_POOL prefix, instead pick closest TYPE_NETWORK
                # .exclude(type=choices.PrefixTypeChoices.TYPE_POOL)
                .get_closest_parent(self.host, include_self=True)
            )
            setattr(self, "_closest_parent", closest_parent)
            return closest_parent
```

Is setattr necessary here or will `self._closest_parent = closest_parent` work?

Reply:
Illd prefer `setattr`, since this class has no attribute `_closest_parent`. Except init is added in the `__init__` method.

Reply:
Please fix the typo `closes_parent` -> `closest_parent`.